crates/bevy_ecs/src/event.rs

```diff
@@ -448,8 +494,12 @@ impl<E: Event> Events<E> {
     }
 
     /// A system that calls [`Events::update`] once per frame.
-    pub fn update_system(mut events: ResMut<Self>) {
-        events.update();
+    pub fn update_system(mut events: ResMut<Self>, settings: Option<Res<EventSettings<E>>>) {
```

Comment:
This is a nice use of an optional resource; I like the way it avoids polluting the resource list with default values.